### DISCUSSION TITLE
Running goreleaser from a different directory to go.mod

### DISCUSSION THREAD

**Discussion by mbamber** [goreleaser/goreleaser#4728] [Q&A] [answered]
Im trying to run `goreleaser` from a different directory to where my source code is located, e.g.

```
├── .goreleaser.yaml
├── gosrc
│   └── main.go
```

My config has `dir: ./gosrc` but the output of `goreleaser release --clean --skip announce,validate,publish` logs this warning/error:
```
• loading go mod information
    • pipe skipped                                   reason=not a go module
```

I've seen #1417 and #2526, the latter implying a bug was fixed (at least in the pro version; I am using the OSS version) so I think I might be misunderstanding something along the way.

Any help greatly appreciated!

**Comment by caarlos0** [comment, marked as answer]
can you test if https://github.com/goreleaser/goreleaser/pull/4729 works?

**Reply by mbamber** [reply]
Hmm I don't think it does unfortunately. Can I just confirm that in order to test I should:
* checkout your branch and run `task build`
* Use the produced `goreleaser` binary e.g. ` ~/workspace/goreleaser/goreleaser release --clean --skip announce,validate,publish` where `~/workspace/goreleaser` is my clone of this repo

**Reply by mbamber** [reply]
Oh wait, I see - I need to set `gomod.dir` now right? If so, then this works perfectly!

**Reply by mbamber** [reply]
Thank you so much for the very speedy response and fix 💯
